### PR TITLE
docs: add documentation for `report_mode` on copy and backport action

### PR DESCRIPTION
**Pull request by Syffe**
Following the implementation of the new report mode config option for backport and copy actions, we needed to add the corresponding documentation.

Fixes MRGFY-3331

### CODE REVIEW

File: public/mergify-configuration-openapi.json [hunk 1]
```diff
@@ -537,6 +537,14 @@
         "$ref": "#/definitions/BranchName"
       }
     },
+    "ReportModeArray": {
+      "$id": "/definitions/ReportModeArray",
+      "description": "list of report modes",
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/ReportModeArray"
```
Comment (jd) [inline public/mergify-configuration-openapi.json:545]: This ref points it this itself so it's a loop.
It should be an enum with the list of possible values.

**Review by jd — DISMISSED** [review]

File: public/mergify-configuration-openapi.json [hunk 2]
```diff
@@ -537,6 +537,12 @@
         "$ref": "#/definitions/BranchName"
       }
     },
+    "ReportModeArray": {
+      "$id": "/definitions/ReportModeArray",
+      "description": "list of report modes",
+      "type": "array",
+      "enum": ["check", "comment"]
```
Comment (jd) [inline public/mergify-configuration-openapi.json:544]: an array cannot be an enum, only a string can be in this case.
So you need to put `items` here and items must be type string and enum with the list
```
"items": {
  "type": "string",
  "enum": ["check", "comment"]
}
```

**Review by jd — CHANGES_REQUESTED** [review]
double check what I'm saying but that should be good

File: public/mergify-configuration-openapi.json (hunk 2 above)
Comment (Syffe) [reply public/mergify-configuration-openapi.json]: Indeed good catch, I didn't knew how to put it, now I know!

**Review by lecrepont01 — DISMISSED** [review]

File: src/content/docs/configuration/data-types.mdx [hunk 3]
```diff
@@ -378,3 +378,32 @@ This describes the reason why a pull request has been removed from the queue.
...
+
+  <tbody>
+    <tr>
+      <td><code>check</code></td>
+      <td>Check is the default report mode. Report the action's result as a GitHub check</td>
+    </tr>
+
+    <tr>
+      <td><code>comment</code></td>
+      <td>report the action's result as a comment on the Pull Request</td>
```
Comment (Greesb) [inline src/content/docs/configuration/data-types.mdx:403]:
```suggestion
      <td>Report the action's result as a comment on the Pull Request</td>
```

File: src/content/docs/configuration/data-types.mdx [hunk 4]
```diff
@@ -378,3 +378,32 @@ This describes the reason why a pull request has been removed from the queue.
...
+<table>
+  <thead>
+    <th>Mode</th>
+    <th>Description</th>
+  </thead>
+
+  <tbody>
+    <tr>
+      <td><code>check</code></td>
+      <td>Check is the default report mode. Report the action's result as a GitHub check</td>
```
Comment (Greesb) [inline src/content/docs/configuration/data-types.mdx:398]:
```suggestion
      <td>Report the action's result as a GitHub check on the Pull Request. This is the default report mode.</td>
```

File: public/mergify-configuration-openapi.json [hunk 5]
```diff
@@ -994,6 +1003,11 @@
               "description": "The pull request title.",
               "default": "{{ title }} (backport #{{ number }})",
               "$ref": "#/definitions/Template"
+            },
+            "report_mode": {
+              "description": "The list of the action's report modes",
```
Comment (Greesb) [inline public/mergify-configuration-openapi.json:1008]:
```suggestion
              "description": "List of reporting modes for the action's result",
```

File: public/mergify-configuration-openapi.json [hunk 6]
```diff
@@ -1049,6 +1063,11 @@
               "description": "The pull request title.",
               "default": "{{ title }} (copy #{{ number }})",
               "$ref": "#/definitions/Template"
+            },
+            "report_mode": {
+              "description": "The list of the action's report modes",
```
Comment (Greesb) [inline public/mergify-configuration-openapi.json:1068]:
```suggestion
              "description": "List of reporting modes for the action's result",
```

**Review by Greesb — DISMISSED** [review]

File: public/mergify-configuration-openapi.json [hunk 7]
```diff
@@ -994,6 +1003,11 @@
               "description": "The pull request title.",
               "default": "{{ title }} (backport #{{ number }})",
               "$ref": "#/definitions/Template"
+            },
+            "report_mode": {
+              "description": "List of reporting modes for the action's result",
```
Comment (jd) [inline public/mergify-configuration-openapi.json:1008]:
```suggestion
              "description": "List of reporting modes for the action's result.",
```

File: public/mergify-configuration-openapi.json [hunk 8]
```diff
@@ -1049,6 +1063,11 @@
               "description": "The pull request title.",
               "default": "{{ title }} (copy #{{ number }})",
               "$ref": "#/definitions/Template"
+            },
+            "report_mode": {
+              "description": "List of reporting modes for the action's result",
```
Comment (jd) [inline public/mergify-configuration-openapi.json:1068]:
```suggestion
              "description": "List of reporting modes for the action's result.",
```

**Review by jd — APPROVED** [review]

**Review by Greesb — APPROVED** [review]

**Pull request merged**